packages/medusa/src/services/event-bus.js

```diff
@@ -163,7 +164,11 @@ class EventBusService {
 
       return stagedJobRepository.save(created)
     } else {
-      this.queue_.add({ eventName, data }, { removeOnComplete: true })
+      const opts = { removeOnComplete: true }
+      if (typeof options.delay === "number") {
+        opts.delay = options.delay
+      }
+      this.queue_.add({ eventName, data }, opts)
```

Comment:
Now that we've added the ability to add options to the queue, wouldn't it make sense to allow for more than just `delay`?

Thinking we could spread the options, and check for `delay`:
``` 
const opts = { ...options, removeOnComplete: true }
if ("delay" in opts && typeof opts.delay !== "number") {
    delete opts.delay
}
```

Reply:
My thinking is that we make a decision on options to include when we migrate the `EventBusService` to TS so that we have a `EmitOptions` type with the specified options.